src/zkevm_specs/evm/execution/balance.py

```diff
@@ -0,0 +1,25 @@
+from ...util import EXTRA_GAS_COST_ACCOUNT_COLD_ACCESS, FQ, N_BYTES_ACCOUNT_ADDRESS
+from ..instruction import Instruction, Transition
+from ..opcode import Opcode
+from ..table import AccountFieldTag, CallContextFieldTag
+
+
+def balance(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+    instruction.constrain_equal(opcode, Opcode.BALANCE)
+
+    address = instruction.rlc_to_fq(instruction.stack_pop(), N_BYTES_ACCOUNT_ADDRESS)
+
+    tx_id = instruction.call_context_lookup(CallContextFieldTag.TxId)
+    is_warm = instruction.add_account_to_access_list(tx_id, address, instruction.reversion_info())
+
+    balance = instruction.account_read(address, AccountFieldTag.Balance)
```

Comment:
not sure if need to constrain "If the given account
doesn't exist, then it will push 0 onto the stack instead." in the circuit , what do you think ? @
silathdiir

Reply:
It seems that the balance is constrained to be equal with `stack_push` in next line:
```
instruction.constrain_equal(instruction.stack_push(), balance)
```
If the given account doesn't exist, `stack_push` should be 0 and also constrain balance to be 0. WDYT? Thanks.

Reply:
After discussing with @DreamWuGit , will send PR to upstream with this question in comment.